include/geos/noding/MCIndexSegmentSetMutualIntersector.h

```diff
@@ -96,15 +105,15 @@ class MCIndexSegmentSetMutualIntersector : public SegmentSetMutualIntersector {
 
 private:
 
-    typedef std::vector<std::unique_ptr<index::chain::MonotoneChain>> MonoChains;
+    typedef std::vector<index::chain::MonotoneChain> MonoChains;
```

Comment:
I had turned `MonoChains` into a vector of vectors (which also preserves stable addresses), but this solution is cleaner and performs better